apps/ui/src/views/Settings/Spaces.vue

```diff
@@ -1,27 +1,37 @@
 <script setup lang="ts">
-import { onMounted } from 'vue';
-
 useTitle('My spaces');
-const { web3Account } = useWeb3();
-const { loaded, spaces, fetch } = useSpaces();
 
-onMounted(() => fetch({ controller: web3Account.value }));
+const { web3 } = useWeb3();
+const { loading: spacesLoading, spaces, fetch } = useSpaces();
+
+const loaded = ref(false);
+
+watch(
+  () => web3.value.account,
+  async controller => {
+    loaded.value = false;
```

Comment:
What's the reason for additional `loaded` flag (on top of the one already in `useSpaces`? Maybe we can come up with better solution that avoids duplication.

Reply:
The one inside useSpaces will be true only after a fetch, which is skipped on guest user, thus always showing the loading icon